File: app/src/main/java/it/amonshore/comikkua/ui/comics/ComicsItemDetails.java
package it.amonshore.comikkua.ui.comics;

import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemDetailsLookup;

public class ComicsItemDetails extends ItemDetailsLookup.ItemDetails<Long> {

    private int mPosition;
    private Long mSelectionKey;

    public ComicsItemDetails(int position, Long selectionKey) {
        mPosition = position;
        mSelectionKey = selectionKey;
    }

    @Override
    public int getPosition() {
        return mPosition;
    }

    @Nullable
    @Override
    public Long getSelectionKey() {
        return mSelectionKey;
    }
}

File: app/src/main/java/it/amonshore/comikkua/ui/comics/adapter/ComicsItemDetails.kt
package it.amonshore.comikkua.ui.comics.adapter

import androidx.recyclerview.selection.ItemDetailsLookup.ItemDetails

class ComicsItemDetails<T>(
    private val position: Int,
    private val selectionKey: T
) :
    ItemDetails<T>() {
    override fun getPosition(): Int {
        return position
    }

    override fun getSelectionKey(): T {
        return selectionKey
    }
}